app/assets/javascripts/labels_index.coffee

```diff
@@ -0,0 +1,183 @@
+class LabelsTable
```

Comment:
I like that you opted for a class here and therefore encapsulated a lot of code - thank you! At first, I was a bit surprised by the fact that you did everything on your own (including the sorting), but I understand that we don't have a nice alternative at the moment. Zuhra is currently working on upgrading bootstrap, which will also allow us to integrate Webpack (by using Shakapacker) and therefore any NPM package. For example, the [`sorttable` package](https://www.npmjs.com/package/sorttable) is the one that we are using in CodeOcean, even though it is already six years old.

Nevertheless, I also considered whether and how this large amount of CoffeeScript could have been reduced / prevented. And actually, since you also extended the sorting to query the server-side objects dynamically, a package such as `sorttable` wouldn't help much (this specific package will well reorder the list, but this state won't be used directly to query new data). 

For modern JavaScript apps, there would be a data binding that would fill the table based on the given data, but this a not yet integrated in CodeHarbor and also wouldn't save too much. The Rails way (as a backend framework) would rather be to render the table using a Slim partial and just swapping the final HTML (probably, this would have been the way I'd choose, but just because I am not a frontend engineer).

Long story short: There are many thoughts I had on this file, but conclusively I think it is fine to leave it as it is. Maybe just take my main concern with you, that we would normally try to leverage the power of libraries and frameworks for those issues that have been solved already many times.

Reply:
I do not think there is a library used in CodeHarbor that could have been used for this and I did not want to use a new one for what I though would be easily solved with some coffeescript. It did end up more complex than I expected though and I will certainly try to use a library/framework next time. Thank you for your feedback!

Reply:
You're welcome, thanks for considering this :)
While a library might have helped here, it also comes at a cost: Just adding a library for every code fragment we need to write is also not an optimal solution -- we need to consider the expected maintenance costs and the increased dependency to something external over the additional complexity involved when writing it our own. So, this one comes with advantages and disadvantages as well...